ablySpec/RealtimeClient.connection.swift

```diff
@@ -713,6 +713,56 @@ class RealtimeClientConnection: QuickSpec {
                 }
 
             }
+
+            // RTN12
+            context("close") {
+
+                // RTN12b
+                it("should receive CLOSED action within the default request timeout") {
+                    let options = AblyTests.commonAppSetup()
+                    options.autoConnect = false
+                    let client = ARTRealtime(options: options)
+                    client.setTransportClass(TestProxyTransport.self)
+                    client.connect()
+
+                    let transport = client.transport as! TestProxyTransport
+                    transport.actionsIgnored += [.Closed]
+
+                    var states: [ARTRealtimeConnectionState] = []
+                    var start: NSDate?
+                    var end: NSDate?
+
+                    client.eventEmitter.on { state, errorInfo in
+                        switch state {
+                        case .Connected:
+                            client.close()
+                        case .Closing:
+                            start = NSDate()
+                            states += [state]
+                        case .Closed:
+                            end = NSDate()
+                            states += [state]
+                        case .Disconnected:
+                            states += [state]
+                        default:
+                            break
+                        }
+                    }
+                    expect(start).toEventuallyNot(beNil(), timeout: testTimeout)
+                    expect(end).toEventuallyNot(beNil(), timeout: ARTDefault.realtimeRequestTimeout())
+
+                    if states.count != 3 {
+                        fail("Invalid number of connection states. Expected CLOSING, DISCONNECTED and CLOSE states")
+                        return
+                    }
+
+                    expect(states[0]).to(equal(ARTRealtimeConnectionState.Closing))
+                    expect(states[1]).to(equal(ARTRealtimeConnectionState.Disconnected))
```

Comment:
There is nothing in the spec that says the connection should move to `Disconnected`, this is wrong.  The connection, when closing, will only ever move to closed directly.